src/policy/block_fees.cpp

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2024 The Bitcoin Core developers
+// Distributed under the MIT software license. See the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <kernel/mempool_entry.h>
+#include <logging.h>
+#include <policy/block_fees.h>
+#include <policy/estimator.h>
+#include <util/trace.h>
+
+#include <queue>
+
+void BlockFees::MempoolTransactionsRemovedForBlock(const std::vector<RemovedMempoolTransactionInfo>& txs_removed_for_block, unsigned int /*unused*/)
+{
+    const std::vector<std::tuple<CFeeRate, uint64_t>> size_per_feerate = LinearizeTransactions(txs_removed_for_block).size_per_feerate;
+    BlockPercentiles percentiles = CalculateBlockPercentiles(size_per_feerate);
+    if (percentiles.p75 != CFeeRate(0)) {
+        if (blocks_percentiles.size() == MAX_NUMBER_OF_BLOCKS) {
+            blocks_percentiles.pop();
+        }
+        blocks_percentiles.push(percentiles);
+    }
+}
+
+ForecastResult BlockFees::EstimateFee(unsigned int targetBlocks)
+{
+    BlockPercentiles percentiles_average;
+
+    if (targetBlocks > BLOCK_FORECAST_MAX_TARGET) {
+        return ForecastResult(strprintf("%s: Confirmation target %u is above the maximum limit of %u. Mempool conditions might change, and estimates above %u are unreliable",
```

Comment:
In  [policy, fees]: add last 6 blocks fee estimate forecaster:

I feel like this makes less sense with a block_fees estimator, but not sure what to suggest to change it to...

Reply:
Yes, updated